generator/internal/language/rust.go

```diff
@@ -776,97 +780,117 @@ func (c *rustCodec) formatDocComments(documentation string, state *api.APIState)
 
 	links := map[string]bool{}
 	var results []string
-	for _, line := range strings.Split(documentation, "\n") {
-		if inBlockquote {
-			switch {
-			case line == "```":
-				inBlockquote = false
-				results = append(results, "```")
-			case strings.HasPrefix(line, blockquotePrefix):
-				results = append(results, strings.TrimPrefix(line, blockquotePrefix))
-			default:
-				inBlockquote = false
+	links := map[string]bool{}
+	md := goldmark.New(
+		goldmark.WithParserOptions(
+			parser.WithAutoHeadingID(),
+		),
+		goldmark.WithExtensions(),
+	)
+
+	documentationBytes := []byte(documentation)
+	doc := md.Parser().Parse(text.NewReader(documentationBytes))
+	ast.Walk(doc, func(node ast.Node, entering bool) (ast.WalkStatus, error) {
+		switch node.Kind() {
+		case ast.KindCodeBlock:
+			if entering {
```

Comment:
These cases are big enough that maybe we want to move them to functions?

Reply:
Done, refactored the logic to leverage helper methods.